libraries/AP_IBus_Telem/AP_IBus_Telem.cpp

```diff
@@ -0,0 +1,193 @@
+                // precaution as it will also be prevented by the PROTOCOL_TIMEGAP required
+                _sensorinfo *s = &_sensors[adr - 1];
+
+                switch (_buffer[0] & 0x0f0) {
+                case PROTOCOL_COMMAND_DISCOVER: { // 0x80, discover sensor
+                    // echo discover command: 0x04, 0x81, 0x7A, 0xFF
+                    struct protocol_command_discover_response_t {
+                        uint8_t command_length;
+                        uint8_t address;
+                        uint16_t checksum;
+                    } packet {
+                        PROTOCOL_FOUR_LENGTH,
+                        (uint8_t)(PROTOCOL_COMMAND_DISCOVER | adr)
+                    };
+                    populate_checksum((uint8_t*)&packet, sizeof(packet));
+                    _port->write((uint8_t*)&packet, sizeof(packet));
+                    break;
+                }
+                case PROTOCOL_COMMAND_TYPE: { // 0x90, send sensor type
+                    // echo sensortype command: 0x06 0x91 0x00 0x02 0x66 0xFF
+                    struct protocol_command_type_response_t {
+                        uint8_t command_length;
+                        uint8_t address;
+                        uint8_t type;
+                        uint8_t length;
+                        uint16_t checksum;
+                    } packet {
+                        PROTOCOL_SIX_LENGTH,
+                        (uint8_t)(PROTOCOL_COMMAND_TYPE | adr),
+                        s->sensor_type,
+                        s->sensor_length
+                    };
+                    populate_checksum((uint8_t*)&packet, sizeof(packet));
+                    _port->write((uint8_t*)&packet, sizeof(packet));
+                    break;
+                }
+                case PROTOCOL_COMMAND_VALUE: { // 0xA0, send sensor data
+                    struct protocol_command_2byte_value_response_t {
+                        uint8_t command_length;
+                        uint8_t address;
+                        uint16_t value;
+                        uint16_t checksum;
+                    } packet {
+                        PROTOCOL_SIX_LENGTH,
+                        (uint8_t)(PROTOCOL_COMMAND_VALUE | adr),
+                        (uint16_t)s->sensor_value
+                    };
+                    populate_checksum((uint8_t*)&packet, sizeof(packet));
+                    _port->write((uint8_t*)&packet, sizeof(packet));
+                    break;
+                }
+                default:
+                    adr = 0; // unknown command, prevent sending chksum
+                    break;
+                }
+            }
+            break;
+        case DISCARD:
+            break;
+        default:
+            break;
+        }
+    }
+}
+
+void AP_IBus_Telem::setSensorMeasurements()
+{
+    //no need to do things if IBusTelemetry is not activated
+    if (_initialized) {
```

Comment:
It's stylistic, but I'd usually do this:
```suggestion
    if (!_initialized) {
        return;
    }
```

This early-return pattern is getting a lot of use in ArduPilot.  In this case you don't need to check if there's code after the `if` block - you know that if `!_initialized` nothing happens.

It's stylistic and you don't need to change this.